runelite-client/src/main/java/net/runelite/client/plugins/opponentinfo/PlayerComparisonOverlay.java

```diff
@@ -0,0 +1,190 @@
+		HiscoreSkill.PRAYER
+	};
+
+	private static final String LEFT_COLUMN_HEADER = "Skill";
+	private static final String RIGHT_COLUMN_HEADER = "You/Them";
+
+	private final Client client;
+	private final OpponentInfoPlugin opponentInfoPlugin;
+	private final OpponentInfoConfig config;
+	private final HiscoreManager hiscoreManager;
+	private final PanelComponent panelComponent = new PanelComponent();
+
+	private HiscoreResult lastOpponentSkills;
+
+	@Inject
+	private PlayerComparisonOverlay(Client client, OpponentInfoPlugin opponentInfoPlugin, OpponentInfoConfig config, HiscoreManager hiscoreManager)
+	{
+		this.client = client;
+		this.opponentInfoPlugin = opponentInfoPlugin;
+		this.config = config;
+		this.hiscoreManager = hiscoreManager;
+
+		setPosition(OverlayPosition.BOTTOM_LEFT);
+		setLayer(OverlayLayer.ABOVE_WIDGETS);
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		if (!config.lookupOnInteraction())
+		{
+			return null;
+		}
+
+		Actor opponent = opponentInfoPlugin.getLastOpponent();
+
+		// Don't try to look up NPC names
+		if (!(opponent instanceof Player))
+		{
+			return null;
+		}
+
+		String opponentName = Text.removeTags(opponent.getName());
+
+		HiscoreResult hiscoreResult = hiscoreManager.lookupAsync(opponentName, opponentInfoPlugin.getHiscoreEndpoint());
+		if (hiscoreResult != null)
+		{
+			lastOpponentSkills = hiscoreResult;
+		}
+
+		if (lastOpponentSkills == null)
+		{
+			return null;
+		}
+
+		panelComponent.getChildren().clear();
+
+		generateComparisonTable(panelComponent);
+
+		return panelComponent.render(graphics);
+	}
+
+	private void generateComparisonTable(PanelComponent panelComponent)
+	{
+		String opponentName = lastOpponentSkills.getPlayer();
+
+		panelComponent.getChildren().add(TitleComponent.builder().text(opponentName).color(HIGHLIGHT_COLOR).build());
+
+		panelComponent.getChildren().add(
+			LineComponent.builder()
+				.left(LEFT_COLUMN_HEADER)
+				.leftColor(HIGHLIGHT_COLOR)
+				.right(RIGHT_COLUMN_HEADER)
+				.rightColor(HIGHLIGHT_COLOR)
+				.build()
+		);
+
+		assert COMBAT_SKILLS.length == HISCORE_COMBAT_SKILLS.length;
+
+		for (int i = 0; i < COMBAT_SKILLS.length; ++i)
+		{
+			Skill skill = COMBAT_SKILLS[i];
+			HiscoreSkill hiscoreSkill = HISCORE_COMBAT_SKILLS[i];
+
+
+			net.runelite.http.api.hiscore.Skill opponentSkill = lastOpponentSkills.getSkill(hiscoreSkill);
+			if (opponentSkill == null || opponentSkill.getLevel() == -1)
+			{
+				continue;
+			}
+
+			int playerSkillLevel = client.getRealSkillLevel(skill);
+			int opponentSkillLevel = opponentSkill.getLevel();
+
+			panelComponent.getChildren().add(
+				LineComponent.builder()
+					.left(hiscoreSkill.getName())
+					.right(Integer.toString(playerSkillLevel) + "/" + Integer.toString(opponentSkillLevel))
+					.rightColor(comparisonStatColor(playerSkillLevel, opponentSkillLevel))
+					.build()
+			);
+		}
+	}
+
+	private static Color comparisonStatColor(int a, int b)
+	{
+		if (a == b)
```

Comment:
remove if/elseif/else chain and make that just ifs, you have early returns in every condition